artiq/protocols/pc_rpc.py

```diff
@@ -487,6 +487,17 @@ def __init__(self, targets, description=None, builtin_terminate=False,
         else:
             self._noparallel = asyncio.Lock()
 
+    @staticmethod
+    def _document_function(function: typing.Callable) -> typing.Tuple[dict, str]:
+        """Turn a function into a tuple of its arguments and documentation.
+        
+        Allows remote inspection of what methods are available on a local device."""
+        argspec_dict = dict(inspect.getfullargspec(function)._asdict())
+        # Fix issue #1186: PYON can't serialize type annotations.
+        if any(argspec_dict.get("annotations", {})):
+            argspec_dict["annotations"] = str(argspec_dict["annotations"])
+        return (argspec_dict, inspect.getdoc(function))
```

Comment:
`return argspec_dict, inspect.getdoc(function)`

Reply:
Fixed in https://github.com/m-labs/artiq/pull/1188/commits/32b0013cc458d92acf4d7243486abcae4595257d